pkgs/tools/networking/tinyproxy/default.nix

```diff
@@ -1,55 +1,36 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook, asciidoc, libxml2,
-  libxslt, docbook_xsl }:
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, perl, withDebug ? false
+, withXTinyProxy ? true, withFilter ? true, withUpstream ? true
+, withTransparent ? true, withReverse ? true, withManpageSupport ? true }:
 
+with lib;
 stdenv.mkDerivation rec {
   pname = "tinyproxy";
-  version = "1.10.0";
+  version = "1.11.0";
 
   src = fetchFromGitHub {
-    sha256 = "0gzapnllzyc005l3rs6iarjk1p5fc8mf9ysbck1mbzbd8xg6w35s";
+    sha256 = "13fhkmmrwzl657dq04x2wagkpjwdrzhkl141qvzr7y7sli8j0w1n";
     rev = version;
     repo = "tinyproxy";
     owner = "tinyproxy";
   };
 
-  nativeBuildInputs = [ autoreconfHook asciidoc libxml2 libxslt docbook_xsl ];
+  nativeBuildInputs = [
+    autoreconfHook
+  ] ++ optionals withManpageSupport [ perl ];
 
-  # -z flag is not supported in darwin
-  preAutoreconf = lib.optionalString stdenv.isDarwin ''
-    substituteInPlace configure.ac --replace \
-          'LDFLAGS="-Wl,-z,defs $LDFLAGS"' \
-          'LDFLAGS="-Wl, $LDFLAGS"'
-  '';
+  configureFlags =
+    optionals    (withDebug)           [ "--enable-debug" ]             # Enable debugging support code and methods.
+    ++ optionals (!withXTinyProxy)     [ "--disable-xtinyproxy" ]       # Compile in support for the XTinyproxy header, which is sent to any web server in your domain.
+    ++ optionals (!withFilter)         [ "--disable-filter"]            # Allows Tinyproxy to filter out certain domains and URLs.
+    ++ optionals (!withUpstream)       [ "--disable-upstream" ]         # Enable support for proxying connections through another proxy server.
+    ++ optionals (!withTransparent)    [ "--disable-transparent" ]      # Allow Tinyproxy to be used as a transparent proxy daemon.
+    ++ optionals (!withReverse)        [ "--disable-reverse" ]          # Enable reverse proxying.
+    ++ optionals (!withManpageSupport) [ "--disable-manpage_support" ]; # Enable support for building manpages.
```

Comment:
Can we just enable all options? That would make the package way less complicated.

Reply:
Would you agree to a toggle just for debugging mode?

Reply:
If that is usefull, we can do that.